### PR TITLE
kernelpkg module

### PR DESCRIPTION
**Pull request by amendlik**
### What does this PR do?
Add execution and state modules to manage Linux kernel packages.
 - Get current kernel version
 - List all installed kernel versions
 - Determine if an upgrade is avaliable in the repos
 - Install kernel upgrades from package repos
 - Determine if a reboot is needed because the active kernel is not the newest one installed

All of these tasks are possible using the pkg module, but the functions are very different between Linux distros. This module abstracts those differences and provides a consistent interface.

YUM and APT are supported in this PR, but other package managers could be added in the future.

### What issues does this PR fix or reference?
None

### Tests written?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

**Comment by damon-atkins** [comment]
I suggest the file name linux_kernelpkg.py   see #30533

File: salt/modules/aptkernelpkg.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+def __virtual__():
+    '''
+    Load this module on Debian-based systems only
+    '''
+
+    if not HAS_REQUIRED_LIBS:
+        return (False, "Required library could not be imported")
+
+    if __grains__.get('os_family') in ('Kali', 'Debian'):
```
Comment (sjorge) [inline salt/modules/aptkernelpkg.py]: Any reason why you do not specify a default on this call like you did on the one on line 33?

**Comment by sjorge** [comment]
@damon-atkins there are a few files here though.

```
states/kernelpkg.py
modules/kernelpkg_linux_apt.py
modules/kernelpkg_linux_yum.py
```

Wouldn't something like this fit better for this one? That leaves room for other pkg managers to be added easily and also allows for other platforms to get there own kernelpkg modules if it makes sense on a later date.


Probably best to move the discussion back to #30533 though, but it would be nice to finally start ironing some of these out more.

**Comment by cachedout** [comment]
Since this is going into develop and we have a while yet to work out the naming question, I am going to merge this. We can deal with the naming issue separately.

**Pull request merged**